tests/conftest.py

```diff
@@ -1,248 +1,284 @@
 import os
 from datetime import datetime
-from decimal import Decimal as PyDecimal
+from decimal import Decimal, DecimalException
 import pytest
-from openpyxl import load_workbook
 from stock_indicators.indicators.common import Quote
+from dateutil.parser import parse, ParserError
 
 dir = os.path.dirname(__file__)
-data_path = os.path.join(dir, "../samples/quotes/History.xlsx")
-wb = load_workbook(data_path, data_only=True)
+
+
+def get_data_from_csv(filename):
+    """Read from CSV file."""
+
+    data_path = os.path.join(dir, f"../samples/quotes/{filename}.csv")
+    with open(data_path, "r") as f:
+        data = f.readlines()
+        data = [d.replace("\n", "").split(",") for d in data]  # parse csv
+    return data[1:]  # skips the first row, those are headers
+
+
+def parse_decimal(value):
+    """Parse decimal value."""
+
+    try:
+        return '{:f}'.format(Decimal(value))
+    except DecimalException:
+        return None
+
 
 @pytest.fixture(scope='session')
 def quotes(days: int = 502):
-    rows = list(wb['Default'])[1:]
+    rows = get_data_from_csv('Default')
 
     h = []
     for row in rows:
         h.append(Quote(
-            row[1].value,
-            row[2].value,
-            row[3].value,
-            row[4].value,
-            row[5].value,
-            row[6].value,
+            parse(row[1]),
+            row[2],
+            row[3],
+            row[4],
+            row[5],
+            row[6],
         ))
 
     h.reverse()
     return h[:days]
 
+
 @pytest.fixture(scope='session')
 def bad_quotes(days: int = 502):
-    rows = list(wb['Bad'])[1:]
+    rows = get_data_from_csv('Bad')
 
     h = []
     for row in rows:
+        ans = None
+        try:
+            ans = parse(row[1])
+        except ParserError:
+            ans = None
```

Comment:
throwing exceptions are better since the error may still pop out if the value is unexpected. for example, passing a non-numeric string "two" into the `parse_decimal` will throw the `DecimalException`.

likewise, for empty values or wrongly formatted values for `parse_date`, will throw value error.
lmkwyt

Reply:
That sounds more reasonable :)